src/sentry/models/release.py

```diff
@@ -170,6 +171,49 @@ def filter_by_semver(
             ).filter(**{f"semver__{semver_filter.operator}": filter_func})
         return qs
 
+    def filter_by_stage(
+        self,
+        organization_id: int,
+        search_filter,
+        project_ids: Sequence[int] = None,
+    ) -> models.QuerySet:
+        from sentry.models import ReleaseProjectEnvironment
+        from sentry.search.events.filter import to_list
+
+        value: str = search_filter.value.value
+        operator: str = search_filter.operator
+        filters = {
+            "adopted": Q(adopted__isnull=False, unadopted__isnull=True),
+            "replaced": Q(adopted__isnull=False, unadopted__isnull=False),
+            "not_adopted": Q(adopted__isnull=True, unadopted__isnull=True),
+        }
+        value = to_list(value)
+        operator_conversions = {"=": "IN", "!=": "NOT IN"}
+        if operator in operator_conversions.keys():
+            operator = operator_conversions[operator]
+
+        for stage in value:
+            if stage not in filters.keys():
```

Comment:
Better off not using `.keys` since it'll give you an iterator, rather than just doing a lookup in the dict